engine/src/main/java/org/pentaho/di/kitchen/KitchenCommandExecutor.java

```diff
@@ -298,13 +300,31 @@ public Job loadJobFromRepository( Repository repository, String dirName, String
     return new Job( repository, jobMeta );
   }
 
-  public Job loadJobFromFilesystem( final String initialDir, final String filename ) throws Exception {
+  public Job loadJobFromFilesystem( String initialDir, String filename, String base64Zip ) throws Exception {
 
     if ( Utils.isEmpty( filename ) ) {
       System.out.println( BaseMessages.getString( getPkgClazz(), "Kitchen.Error.canNotLoadJob" ) );
       return null;
     }
 
+    if ( !Utils.isEmpty( base64Zip ) ) {
```

Comment:
Suggestion: promote this chunk of code to an util class to be re-used in both (...)CommandExecutors classes

Reply:
Agree. I'll issue a follow-up Cleanup PR promoting what I can, for re-use at Pan and Kitchen

Reply:
Follow up PR will be part of BACKLOG-28228?